retworkx-core/src/planar/lr_planar.rs

```diff
@@ -0,0 +1,699 @@
+                .edge_references()
+                .map(|e| ((e.source(), e.target()), Sign::Plus))
+                .collect(),
+        }
+    }
+
+    fn lr_orientation_visitor(&mut self, event: DfsEvent<G::NodeId, &G::EdgeWeight>) {
+        match event {
+            DfsEvent::Discover(v, _) => {
+                if let Entry::Vacant(entry) = self.height.entry(v) {
+                    entry.insert(0);
+                    self.roots.push(v);
+                }
+            }
+            DfsEvent::TreeEdge(v, w, _) => {
+                let ei = (v, w);
+                let v_height = self.height[&v];
+                let w_height = v_height + 1;
+
+                self.eparent.insert(w, ei);
+                self.height.insert(w, w_height);
+                // now initialize low points.
+                self.lowpt.insert(ei, v_height);
+                self.lowpt_2.insert(ei, w_height);
+            }
+            DfsEvent::BackEdge(v, w, _) => {
+                // do *not* consider ``(v, w)`` as a back edge if ``(w, v)`` is a tree edge.
+                if Some(&(w, v)) != self.eparent.get(&v) {
+                    let ei = (v, w);
+                    self.lowpt.insert(ei, self.height[&w]);
+                    self.lowpt_2.insert(ei, self.height[&v]);
+                }
+            }
+            DfsEvent::Finish(v, _) => {
+                for edge in self.graph.edges(v) {
+                    let w = edge.target();
+                    let ei = (v, w);
+
+                    // determine nesting depth.
+                    let low = match self.lowpt.get(&ei) {
+                        Some(val) => *val,
+                        None =>
+                        // if ``lowpt`` does *not* contain edge ``(v, w)``, it means
+                        // that it's *not* a tree or a back edge so we skip it since
+                        // it's oriented in the reverse direction.
+                        {
+                            continue
+                        }
+                    };
+
+                    if self.lowpt_2[&ei] < self.height[&v] {
+                        // if it's chordal, add one.
+                        self.nesting_depth.insert(ei, 2 * low + 1);
+                    } else {
+                        self.nesting_depth.insert(ei, 2 * low);
+                    }
+
+                    // update lowpoints of parent edge.
+                    if let Some(e_par) = self.eparent.get(&v) {
+                        match self.lowpt[&ei].cmp(&self.lowpt[e_par]) {
+                            Ordering::Less => {
+                                self.lowpt_2
+                                    .insert(*e_par, self.lowpt[e_par].min(self.lowpt_2[&ei]));
+                                self.lowpt.insert(*e_par, self.lowpt[&ei]);
+                            }
+                            Ordering::Greater => {
+                                insert_or_min(&mut self.lowpt_2, *e_par, self.lowpt[&ei]);
```

Comment:
Yes it's equivalent but not necessarily faster since it evaluates the hash of `e_par` twice (ignoring any compiler optimization): 
1. retrieve the value `lowpt_2[e_par]`, and 
2. insert the updated value `lowpt_2.insert(*e_par, ..)`

But you've got a point: `e_par` is already contained in `lowpt_2` and hence the `or_insert` path in `insert_or_min` function can be omitted. I've done this in 0ef2844